### PR TITLE
Created event_listener_fourth rule

### LINKED ISSUE #353: Rule: addEventListener_fourth
https://github.com/mozilla/addons-validator/blob/master/docs/rules.md#instanceactions

### PR DESCRIPTION
**Pull request by Nolski**
Resolves #353

### CODE REVIEW

File: tests/rules/javascript/test.event_listener_fourth.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import JavaScriptScanner from 'scanners/javascript';
+import { VALIDATION_WARNING } from 'const';
+import { singleLineString } from 'utils';
+import * as messages from 'messages';
+
+describe('event_listener_fourth', () => {
+  it('should not allow a true literal', () => {
+    var code = 'window.addEventListener("click", function(){}, false, true);';
```
Comment (muffinresearch) [inline tests/rules/javascript/test.event_listener_fourth.js]: Nice comprehensive tests.

I wonder if we could add something to someone using `window.addEventListener.call` and `window.addEventListener.apply`. Presumably the old code doesn't go that far?

**Comment by muffinresearch** [comment]
r+wc

File: tests/rules/javascript/test.event_listener_fourth.js (hunk 1 above)
Comment (Nolski) [inline tests/rules/javascript/test.event_listener_fourth.js]: I agree, in fact, I'm beginning to wonder if there may be a better way to wrap ESLint here. For instance when we are looking for call expressions for something like `foo()` we normally are looking for `window.foo()` or `var a = foo; a();` or other sort of things. And I'm seeing that the rules are often having to contain this logic individually and some of it is starting to get repeated.

Comment (muffinresearch) [inline tests/rules/javascript/test.event_listener_fourth.js]: Yep the nicest thing I think we can do here is keep the many rules but encapsulate more functionality into the utils helpers. That way we can keep things dry but also retain as much of the eslint way of doing things as possible.

**Pull request merged**